.github/workflows/ci.yaml

```diff
@@ -40,20 +40,30 @@ jobs:
       - name: Install ffigen
         run: dart pub global activate ffigen ^5.0.0 # TODO migrate to 6.x later
 
-      - name: Install dart dependencies
+      - name: Install dart dependencies (single block)
         working-directory: ./frb_example/pure_dart/dart
         run: dart pub get
 
-      - name: Run valgrind script
+      - name: Install dart dependencies (multi blocks)
+        working-directory: ./frb_example/pure_dart_multi/dart
+        run: dart pub get
+
+      - name: Run bash and valgrind script (single block)
         working-directory: ./frb_example/pure_dart/dart
         run: chmod +x ./run.sh ./valgrind_util.py && ./run.sh
         env:
           CARGO_TARGET_DIR: /home/runner # NOTE here, must be an absolute path, since in my script uses it in multiple different directories
 
+      - name: Run bash script (multi blocks)
+        working-directory: ./frb_example/pure_dart_multi/dart
+        run: chmod +x ./run.sh
```

Comment:
Hey this is wrong! You only `chmod`, i.e. add permission to file, but never execute it.

What about `dart test` or something like that.

Reply:
> Hey this is wrong! You only `chmod`, i.e. add permission to file, but never execute it.

Sorry for missing command, I've fixed in cc549d25ec2bffe128a3139952f1290c0be20cf7.

> What about `dart test` or something like that.
There is already some testing related to `pure_dart_multi `:
```yaml
      - name: Install dependencies for pure_dart_multi example
        working-directory: ./frb_example/pure_dart_multi/dart
        run: dart pub get
      - name: Verify formatting for pure_dart_multi example
        working-directory: ./frb_example/pure_dart_multi/dart
        run: dart format --output=none --set-exit-if-changed --line-length 120 .
      - name: Analyze project source for pure_dart_multi example
        working-directory: ./frb_example/pure_dart_multi/dart
        run: dart analyze --fatal-infos
```
You can see more tests by keyword `pure_dart_multi`, still not enough?

Reply:
`format` and `analyze` are **NOT TESTS**.

Reply:
> `format` and `analyze` are **NOT TESTS**.

I've no idea what other tests should be done here. Currently, there is a `run.sh`:
```sh
#!/bin/bash

set -eux

if [[ -z "${CARGO_TARGET_DIR}" ]]; then
  echo 'Please set environment variable CARGO_TARGET_DIR'
  exit 1
fi

(cd ../rust && cargo build --verbose)

# dart pub get

# need to be AOT, since prod environment is AOT, and JIT+valgrind will have strange problems
dart compile exe lib/main.dart -o main.exe
```

Is there something else need to add?

Reply:
That is nothing but *compile* things, but not *test* things.

Maybe firstly learn a bit about how people test things in dart/flutter, and then you will know it

Reply:
> That is nothing but _compile_ things, but not _test_ things.
> 
> Maybe firstly learn a bit about how people test things in dart/flutter, and then you will know it

There is already  some test code here `pure_dart_multi\dart\lib\main.dart`:
```dart
 test('dart call simpleAdder', () async {
    expect(await api1.simpleAdder1(a: 42, b: 100), 142);
  });

  final dylib2 = DynamicLibrary.open(dylibPath2);
  final api2 = ApiClass2Impl(dylib2);
  test('dart call simpleAdder2', () async {
    expect(await api2.simpleAdder2(a: 42, b: 100), 142);
  });
```
Do you mean there is an essential to explicitly add a test command, like `dart test ...`?

Reply:
Yes. run command like `dart test` will execute it.

Reply:
> Yes. run command like `dart test` will execute it.

May I ask, why there is no such test command for the original example `pure_dart\dart`?

Reply:
Because it compiled `.dart` into an executable binary, and use Valgrind to run that binary (with a small python wrapper script), because we want Valgrind to detect possible memory problems.